lib/src/controls/flyouts/content_manager.dart

```diff
@@ -0,0 +1,124 @@
+import 'package:fluent_ui/fluent_ui.dart';
+
+class ContentManager extends StatefulWidget {
+  const ContentManager({
+    Key? key,
+    required this.content,
+  }) : super(key: key);
+
+  final WidgetBuilder content;
+
+  @override
+  State<ContentManager> createState() => _ContentManagerState();
+}
+
+class _ContentManagerState extends State<ContentManager> {
+  final GlobalKey key = GlobalKey();
+  Size size = Size.zero;
+
+  @override
+  void initState() {
+    WidgetsBinding.instance.addPostFrameCallback((timeStamp) {
+      final context = key.currentContext;
+      if (context == null) return;
+      final box = context.findRenderObject() as RenderBox;
+      setState(() => size = box.size);
+    });
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ContentSizeInfo(
+      size: size,
+      child: Builder(builder: (context) {
+        return KeyedSubtree(
+          key: key,
+          child: widget.content(context),
+        );
+      }),
+    );
+  }
+}
+
+class ContentSizeInfo extends InheritedWidget {
+  const ContentSizeInfo({
+    Key? key,
+    required Widget child,
+    required this.size,
+  }) : super(key: key, child: child);
+  final Size size;
+  static ContentSizeInfo of(BuildContext context) {
+    return context.dependOnInheritedWidgetOfExactType<ContentSizeInfo>()!;
+  }
+
+  static ContentSizeInfo? maybeOf(BuildContext context) {
+    return context.dependOnInheritedWidgetOfExactType<ContentSizeInfo>();
+  }
+
+  @override
+  bool updateShouldNotify(ContentSizeInfo oldWidget) {
+    return oldWidget.size != size;
+  }
+}
+
+typedef _MenuBuilder = Widget Function(
+  BuildContext context,
+  BoxConstraints rootSize,
+  Iterable<Widget> menus,
+  Iterable<GlobalKey> keys,
+);
+
+@protected
```

Comment:
This annotation here makes no sense, works only on instance members, not on classes themselves